app/src/main/java/org/kiwix/kiwixmobile/webserver/ZimHostModule.java

```diff
@@ -0,0 +1,23 @@
+package org.kiwix.kiwixmobile.webserver;
+
+import dagger.Module;
+import dagger.Provides;
+import javax.inject.Singleton;
+import org.kiwix.kiwixmobile.di.PerActivity;
+import org.kiwix.kiwixmobile.utils.AlertDialogShower;
+
+@Module
+public class ZimHostModule {
+
+  @PerActivity
+  @Provides
+  ZimHostContract.Presenter provideZimHostPresenter(ZimHostPresenter zimHostPresenter) {
+    return zimHostPresenter;
+  }
+
+  @Provides
+  @Singleton AlertDialogShower provideAlertDialogShower(AlertDialogShower alertDialogShower) {
```

Comment:
When providing dependencies via dagger and the class you want to provide implements an interface it is nicer to Provide the interface, so
```
DialogShower provideAlertDialogShower(AlertDialogShower alertDialogShower) {
```

anf then when you inject
```
@Inject DialogShower dialogShower;
```
and that variable receives the instance of AlertDialogShower.
/@Singleton is also not correct, /@PerActivity is the way to go

Reply:
Fixed. fc4ba04